crates/walrus-service/src/node/node_recovery.rs

```diff
@@ -133,10 +145,25 @@ impl NodeRecoveryHandler {
                         while ongoing_syncs.len() >= max_concurrent_blob_syncs_during_recovery {
                             ongoing_syncs.next().await;
                         }
+
+                        // Since there is a wait, the blob might not be certified anymore. Check
```

Comment:
In addition to @mlegner 's change to watch for blob retirement inside BlobSyncHandler, adding another check here to avoid unnecessary tasks started.

Reply:
Yes, and if we want to do the check right before starting blob sync, the sliver check also needs to be moved. Let me tackle this in a follow up PR as part of WAL-864